crates/sui-types/src/event.rs

```diff
@@ -131,8 +127,8 @@ impl Event {
         resolver: &impl GetModule,
     ) -> Result<Option<MoveStruct>, SuiError> {
         match self {
-            Event::MoveEvent { type_, contents } => {
-                let typestruct = TypeTag::Struct(type_.clone());
+            Event::MoveEvent(event) => {
```

Comment:
maybe call the inside `event_obj` or something ... just to clarify it's actually a `MoveObject`?